service/src/main/kotlin/io/provenance/onboarding/frameworks/cee/P8eContractService.kt

```diff
@@ -26,26 +30,61 @@ class P8eContractService : ContractService {
         contractClass: Class<T>,
         records: Map<String, Message>,
         scopeUuid: UUID,
-        sessionUuid: UUID?
+        sessionUuid: UUID?,
+        participants: Map<Specifications.PartyType, Originator>?,
+        scope: ScopeResponse?,
     ): Session =
-        client
-            .newSession(contractClass, LoanScopeSpecification::class.java)
-            .setScopeUuid(scopeUuid)
-            .configureSession(records, sessionUuid)
-            .also { session ->
-                log.info("[L: ${session.scopeUuid}, S: ${session.sessionUuid}] ${contractClass.simpleName} has been setup.")
-            }
+        when (scope) {
+            null ->
+                client
+                    .newSession(contractClass, LoanScopeSpecification::class.java)
+                    .setScopeUuid(scopeUuid)
+                    .configureSession(records, sessionUuid, participants)
+                    .also { session ->
+                        log.info("[L: ${session.scopeUuid}, S: ${session.sessionUuid}] ${contractClass.simpleName} has been setup.")
+                    }
+            else ->
+                client
+                    .newSession(contractClass, scope)
+                    .configureSession(records, sessionUuid, participants)
+                    .also { session ->
+                        log.info("[L: ${session.scopeUuid}, S: ${session.sessionUuid}] ${contractClass.simpleName} has been setup.")
+                    }
+        }
+
+    override fun executeContract(client: Client, session: Session): ExecutionResult =
+        runCatchingExecutionResult<Session>() {
+            client.execute(session)
+        }
+
+    override fun executeContract(client: Client, envelope: Envelope): ExecutionResult =
+        runCatchingExecutionResult<Envelope>() {
+            client.execute(envelope)
+        }
 
-    override fun executeContract(client: Client, session: Session, executeTransaction: (SingleTx) -> Abci.TxResponse): Result<Abci.TxResponse> =
+    private fun <T> runCatchingExecutionResult(execution: () -> ExecutionResult): ExecutionResult =
         runCatching {
-            when (val result = client.execute(session)) {
-                is SignedResult -> executeTransaction(SingleTx(result))
-                else -> throw IllegalStateException("Must be a signed result since this is a single party contract.")
+            execution()
+        }.fold(
+            onSuccess = { result ->
+                return result
+            },
+            onFailure = { throwable ->
+                throw throwable
             }
-        }
+        )
```

Comment:
Nit: Is this eventually going to do something else, like wrap the exception with one defined in this project? Otherwise, I am confused as to what the `runCatching` (and possibly this function itself) are useful for.

Reply:
fair. I updated this to throw an appropriate exception `ContractExecutionException`. Honestly, we could probably refactor this out in the future if we wanted to streamline even more.